### ISSUE TITLE
Add comments and executable comments to RBScanner class methods

### ISSUE THREAD

**Issue by Ducasse** [guillep/pharo-core#5451]
Why are we forced to read the code while a good comment and executable comment would do the job.
Here is a simple example of how we can welcome more users. 

```
RBScanner class >> on: aStream 
	| str |
	str := self new on: aStream.
	str
		step;
		stripSeparators.
	^str
```

```
RBScanner class >> on: aStream 
	"Create a scanner stream of program tokens on the argument. Users can then use stream API to consume tokens. Reader may consider scanTokens: as a better way to use the class."
        "((RBScanner on: 'Pharo (, { exist ]' readStream) next;  next) value >>> $("

        | str |
	str := self new on: aStream.
	str
		step;
		stripSeparators.
	^str
```

**Comment by MarcusDenker** [comment]
But we should make it clear, too, that the Scanner should not be used directly if it can be avoided. (it is an implementation detail of that specific Parser...)

**Comment by olekscode** [comment]
> But we should make it clear, too, that the Scanner should not be used directly if it can be avoided. (it is an implementation detail of that specific Parser...)

I think this information should be added to the class comment of `RBScanner`, not to `RBScanner >> on:`. @MarcusDenker what do you think?

**Comment by MarcusDenker** [comment]
yes, class comment is good

**Comment by olekscode** [comment]
This issue should be closed because the PR #6447 has been merged.
I have created a separate issue for improving the class comment: #6470

**Comment by MarcusDenker** [comment]
Ok!
